### DISCUSSION TITLE
Authentication failed, credentials for `(not available)` are not valid

### DISCUSSION THREAD

**Discussion by xROAL** [prisma/prisma#11500] [Q&A]
Running any of the `migrate`, `db pull` or `db push` commands gives me this error message:

```
Environment variables loaded from .env
Prisma schema loaded from prisma\schema.prisma
Datasource "db": PostgreSQL database "k838n1m6", schema "public" at "sub.domain.tld:5432"

Error: P1000: Authentication failed against database server at `sub.domain.tld`, the provided database credentials for `(not available)` are not valid.

Please make sure to provide valid database credentials for the database server at `sub.domain.tld`.
error Command failed with exit code 1.
```

Both my username and my password contain only alphanumeric characters (`[a-zA-Z0-9]`) and I am able to connect to the server using tools such as DBeaver from the same machine without any issues, I can copy/paste the username, password, host and everything from the connection string I'm giving to Prisma and it works.

It seems though that Prisma is not even picking up the username, as it shows `not available` where username should be in the error message.

I've got no idea what to try, or where to look for more information. Any ideas, please?

I'm using `prisma` and `@prisma/client` in version `3.8.1` with Node `16.13.2` on Windows 10, the database is PostgreSQL 10.0 and is accessible publicly.

**Comment by janpio** [comment]
What does your connection string look like? That it can not pick that up is weird.

**Reply by xROAL** [reply]
The content of my `.env` file is:

```
DATABASE_URL="postgresql://k838n1m6:pwd@sub.domain.tld:5432/k838n1m6?schema=public&application_name=myapp"
```

In my `prisma.schema`, I have:

```
datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}
```

The name of the database, as well as the host and port are shown correctly in the output of those commands, it's just the username that is missing.

**Reply by xROAL** [reply]
I've just created a local database in Docker with the latest `postgres` image and everything works just fine with the following connection string:

```
DATABASE_URL="postgresql://postgres:pwd@localhost:5432/myappdb?schema=public&application_name=myapp"
```

There seems to be some strange problem with that specific connection string, so I tried to create a new database with a different user/password on the same provider, but no change.

When I get around to it, I will try to create a database on some other provider and see if the issue persists.

Is it feasible that something on the provider side would be causing this?

**Reply by janpio** [reply]
Yes, _something_ might be weird - but this is really hard to debug (as you can see with the error message). Can you share which provider you are using?

**Reply by xROAL** [reply]
I tried to create an instance of Postgres DB on Heroku and everything works just fine.

Originally, I was using a DB on a shared hosting provider. Might be related to user rights/roles then, maybe.

Anyways I think this confirms that there's an issue on their side, so I guess we can close this.

**Reply by janpio** [reply]
Can you share the hosting provider? We optimally want Prisma to work everywhere, so this might be worth investigating.

**Reply by xROAL** [reply]
It is a Slovak-based shared hosting provider, as far as I know, they operate in Slovakia, Czechia, and Hungary. While they definitely have English-speaking staff, navigating their site might be a bit harder, I'm not sure whether they have an English version.

https://www.websupport.sk/

The contact page is at https://www.websupport.sk/kontakt/

**Reply by janpio** [reply]
Google translate works surprisingly well for these cases :D 

The "Mini" web hosting package would probably be enough?

**Reply by xROAL** [reply]
Glad it works. I think that should definitely be enough.

**Reply by xROAL** [reply]
Another note - as they offer other types of databases as well, I've created a MariaDB 10.5 with them, and using the `mysql` connector, everything works, so it seems it's only Postgre on their side that is causing some issues.

**Comment by reorx** [comment]
I encountered the same error, I'm using a database from bit.io, the connection url is:

```
postgresql://reorx:pwd@db.bit.io/reorx/umami
```

I think bit.io is a good case to test against prisma client.

**Reply by janpio** [reply]
Interesting, we have not encountered this service before. Can you please open a real bug issue for this? Then we can track our investigation what is going on there more easily. Thanks!

**Reply by reorx** [reply]
Just found this was caused by the database name `reorx/umami` is not accepted by prisma, bit.io clarified it in their documentation: https://docs.bit.io/docs/connecting-via-prisma#connecting-to-bitio-with-prisma

After changing `reorx/umami` to `reorx.umami`, I could successfully connect to the database via prisma :)

I'm not very familiar with postgres, so I don't know which one is closer to the specification of database url.

**Reply by reorx** [reply]
Though my problem is resolved, I do think prisma could provide a clearer error message, I can never realize it was a format problem with the message of "Authentication failed"

**Reply by janpio** [reply]
If database names are allowed to have `/` in them by the standards, then this should totally not be a problem and this is a bug on our side we need to fix. Can you please open a bug report issue for this? That would be super helpful for us to track this. Thanks!

**Reply by glauber-cruz** [reply]
Thanks dude, ur answer help me out a lot to solve the same problem lol I spend 4 f.cking hours trying to find out this, god dammit, prisma doesnt even give ya a clue that this was the problem OMG

**Reply by janpio** [reply]
I am not aware of Prisma having a bug report about this. Please open one and describe the full problem.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
